FavoriteArtists/FavoriteArtists/ArtistPersistenceController.m

```diff
@@ -0,0 +1,87 @@
+//
+//  ArtistPersistenceController.m
+//  FavoriteArtists
+//
+//  Created by Clayton Watkins on 9/25/20.
+//
+
+#import "ArtistPersistenceController.h"
+#import "Artist.h"
+#import "Artist+JSONSerialization.h"
+
+
+@interface ArtistPersistenceController () {
+    NSMutableArray *_internalArtists;
+}
+
+@end
+
+@implementation ArtistPersistenceController
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        _internalArtists = [[NSMutableArray alloc] init];
+        
+        // Load favorite artists from disk
+        NSString *filePath = self.getFilePath;
+        NSFileManager *fileManager = [NSFileManager defaultManager];
+        
+        if ([fileManager fileExistsAtPath:filePath]) {
+            NSDictionary *dictionary = [[NSDictionary alloc] initWithContentsOfURL:self.getFileURL];
+            [self updateArtistsArrayWithDictionary:dictionary];
+        }
+    }
+    return self;
+}
+
+#pragma mark - Convenience Accessors
+
+- (NSArray<Artist *> *)artists
+{
+    return _internalArtists.copy;
+}
+
+- (void)addArtist:(Artist *)artist
+{
+    [_internalArtists addObject:artist];
+    [self saveToPersistentStore];
+}
+
+- (NSURL *)getFileURL
+{
+    NSString *fileName = @"FavoriteArtistsData.plist";
+    NSString *path = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject];
```

Comment:
Consider using https://developer.apple.com/documentation/foundation/nsfilemanager/1407693-urlfordirectory?language=occ instead to grab a NSURL directly.

Reply:
```suggestion
    NSString *path = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES).firstObject;
```